gateway/src/main/kotlin/DefaultGateway.kt

```diff
@@ -187,8 +181,37 @@ public class DefaultGateway(private val data: DefaultGatewayData) : Gateway {
         }
 
         try {
-            defaultGatewayLogger.trace { "Gateway <<< $json" }
-            val event = jsonParser.decodeFromString(Event.DeserializationStrategy, json) ?: return
+            val event = GatewayJson.decodeFromString(Event.DeserializationStrategy, json)
+
+            defaultGatewayLogger.trace {
+                val credentialFreeJson = when (event) {
+
+                    is VoiceServerUpdate -> {
+                        when (val payload = GatewayJson.parseToJsonElement(json)) {
+                            is JsonObject -> {
+                                val payloadCopy = buildJsonObject {
+                                    for ((k, v) in payload) put(k, v)
+
+                                    val data = payload["d"]
+                                    if (data is JsonObject) putJsonObject("d") {
+                                        for ((k, v) in data) put(k, v)
+                                        put("token", "hunter2")
+                                    }
+                                }
+                                payloadCopy.toString()
+                            }
+                            else -> json
+                        }
+                    }
+
+                    else -> json
+                }
+
+                "Gateway <<< $credentialFreeJson"
+            }
```

Comment:
This is quite heavy for just not logging the token, voice does it like [this](https://github.com/kordlib/kord/blob/bebfcc74efae44d2e578b6d90076f1265eb9e049/voice/src/main/kotlin/gateway/DefaultVoiceGateway.kt#L149) to hide [`secret_key`](https://discord.com/developers/docs/topics/voice-connections#establishing-a-voice-udp-connection-example-session-description-payload), should we do it like that too instead?